engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/MetaDataBuilder.java

```diff
@@ -73,7 +75,11 @@ public void add(ConstrainedElement constrainedElement) {
 		constraints.addAll( adaptConstraints( constrainedElement.getKind(), constrainedElement.getConstraints() ) );
 		constraints.addAll( adaptConstraints( constrainedElement.getKind(), new HashSet<>( constrainedElement.getTypeArgumentConstraints().values() ) ) );
 		isCascading = isCascading || constrainedElement.isCascading();
-		unwrapMode = constrainedElement.unwrapMode();
+
+		if ( unwrapMode == null || constrainedElement.getSource().getPriority() > unwrapModeSource.getPriority() ) {
+			unwrapMode = constrainedElement.unwrapMode();
+			unwrapModeSource = constrainedElement.getSource();
```

Comment:
Yes, it's a bug fix; it was uncovered in the course of implementing the new mechanism. I wouldn't bother backporting unless a user reports it.